bazel/rules_go.patch

```diff
@@ -12,6 +12,18 @@
 # - https://github.com/bazelbuild/rules_go/pull/2542
 # - https://github.com/envoyproxy/envoy/issues/11657
 #
+diff --git a/go/private/platforms.bzl b/go/private/platforms.bzl
+index a16e2993..21a62e90 100644
+--- a/go/private/platforms.bzl
++++ b/go/private/platforms.bzl
+@@ -30,7 +30,6 @@ BAZEL_GOARCH_CONSTRAINTS = {
+     "amd64": "@platforms//cpu:x86_64",
+     "arm": "@platforms//cpu:arm",
+     "arm64": "@platforms//cpu:aarch64",
+-    "ppc64": "@platforms//cpu:ppc",
```

Comment:
Why can't we upstream and we have to resort to patches?

Reply:
@htuch Two issues are still open with the rules_go community, hence the delay:
https://github.com/bazelbuild/rules_go/issues/3626
https://github.com/bazelbuild/rules_go/issues/3531

Reply:
also making some progress there. apparently the big and little endian cpu types were using the same platform constraint. this was just merged: https://github.com/bazelbuild/platforms/pull/64

Reply:
> also making some progress there. apparently the big and little endian cpu types were using the same platform constraint. this was just merged: [bazelbuild/platforms#64](https://github.com/bazelbuild/platforms/pull/64)

@yanavlasov @htuch We applied a patch to bazel 6.3.2 and rules_go (added `"ppc64le": "@platforms//cpu:ppc64le"`), then tried to build envoy, but somehow the cpu addition is not taking effect. Can you tell us where the envoy build is picking up the bazelbuild/platforms version? cc: @seth-priya @clnperez

Reply:
You can refer to what I changed several years ago to get envoy to build for Power: https://github.com/envoyproxy/envoy/blob/main/bazel/BUILD
though I'm not sure if some of that has been reworked since then.

Reply:
I think @keith @phlax have better coverage these days on these kinds of Bazel issues.